### DISCUSSION TITLE
Using transmit-client with React-Native App

### DISCUSSION THREAD

**Discussion by rhnkyr** [adonisjs/core#4488] [Help] [answered]
Hi,

is there a way to make transmit-client library to work with react-native?

**Comment by Julien-R44** [comment, marked as answer]
I think, React Native doesnt have a builtin EventSource API

So what you can do is use this package https://github.com/binaryminds/react-native-sse which provides an EventSource for react native, and pass it in when creating your transmit client instance

```ts
import EventSource from "react-native-sse";

const transmit = new Transmit({
  baseUrl: 'http://localhost:3333/',
  eventSourceConstructor: EventSource
})
```

And it should work fine

**Comment by rhnkyr** [comment]
@Julien-R44 thank you!

**Comment by rhnkyr** [comment]
@Julien-R44 this approach throws "Invalid Package Error" :/ Any idea?

The stack trace:

`Metro has encountered an error: While trying to resolve module ⁠ @adonisjs/transmit-client ⁠ from file ⁠ /Users/user/Documents/Projects/Meyouw/App/pages/launch/index.js ⁠, the package ⁠ /Users/user/Documents/Projects/Meyouw/node_modules/@adonisjs/transmit-client/package.json ⁠ was successfully found. However, this package itself specifies a ⁠ main ⁠ module field that could not be resolved (⁠ /Users/user/Documents/Projects/Meyouw/node_modules/@adonisjs/transmit-client/build/transmit.js ⁠. Indeed, none of these files exist:

  * /Users/user/Documents/Projects/Meyouw/node_modules/@adonisjs/transmit-client/build/transmit.js(.native|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx)
  * /Users/user/Documents/Projects/Meyouw/node_modules/@adonisjs/transmit-client/build/transmit.js/index(.native|.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx): /Users/user/Documents/Projects/Meyouw/node_modules/metro/src/node-haste/DependencyGraph.js (289:17)

  287 |         }
  288 |         if (error instanceof InvalidPackageError) {
    ⁠289 |           throw new PackageResolutionError({
      |                 ^
  290 |             packageError: error,
  291 |             originModulePath: from,
  292 |             targetModuleName: to,

RCTFatal
__28-[RCTCxxBridge handleError:]_block_invoke
_dispatch_call_block_and_release
_dispatch_client_callout
_dispatch_main_queue_drain
_dispatch_main_queue_callback_4CF
_CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE_
__CFRunLoopRun
CFRunLoopRunSpecific
GSEventRunModal
-[UIApplication _run]
UIApplicationMain
main
start_sim
0x0`
